doc/nrf/device_guides/working_with_nrf/nrf53/qspi_xip_guide.rst

```diff
@@ -0,0 +1,352 @@
+.. _qspi_xip:
+
+Configuring external XIP
+########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This guide describes the external execute in place (XIP) support for the nRF5340 SoC.
+The nRF5340 SoC is equipped with a Quad Serial Peripheral Interface (QSPI) memory interface, which is capable of exposing QSPI flash as memory for the CPU used to fetch and execute the program code.
+The QSPI is available for the application core.
+Therefore, it is possible to relocate a part of the application's code to an external memory.
+NCS supports dividing an application into an internal and external part, along with the MCUboot support.
+
+To configure a project for QSPI XIP, you need to perform the following steps:
+
+ * Provide the QSPI flash memory DTS.
+ * Provide a dedicated static partition manager configuration file with proper internal and external partitioning.
+ * Provide a linker file that describes the memory block of the QSPI XIP code location.
+ * Describe code relocation in the project's CMake file.
+
+Additionally, you need to set up regular project elements, for example, an application for the application core and a networking stack for the networking core.
```

Comment:
@nvlsianpu do you think that maybe we could drop these lines completely? I feel like it repeats the headings that will be displayed in the navigation menu on the side.